pytorch_lightning/loggers/neptune.py

```diff
@@ -46,235 +98,437 @@ class NeptuneLogger(LightningLoggerBase):
         """
-        # Neptune does not save any local files
-        return None
+        return os.path.join(os.getcwd(), ".neptune")
+
+    def log_model_summary(self, model, max_depth=-1):
+        model_str = str(ModelSummary(model=model, max_depth=max_depth))
+        self.experiment[
+            self._construct_path_with_prefix("model/summary")
+        ] = neptune.types.File.from_content(content=model_str, extension="txt")
+
+    def after_save_checkpoint(self, checkpoint_callback: "ReferenceType[ModelCheckpoint]") -> None:
+        """
+        Called after model checkpoint callback saves a new checkpoint
+
+        Args:
+            checkpoint_callback: the model checkpoint callback instance
+        """
+        if not self._log_model_checkpoints:
+            return
+
+        file_names = set()
+        checkpoints_namespace = self._construct_path_with_prefix("model/checkpoints")
+
+        # save last model
+        if checkpoint_callback.last_model_path:
+            model_last_name = self._get_full_model_name(checkpoint_callback.last_model_path, checkpoint_callback)
+            file_names.add(model_last_name)
+            self.experiment[f"{checkpoints_namespace}/{model_last_name}"].upload(
+                checkpoint_callback.last_model_path
+            )
+
+        # save best k models
+        for key in checkpoint_callback.best_k_models.keys():
+            model_name = self._get_full_model_name(key, checkpoint_callback)
+            file_names.add(model_name)
+            self.experiment[f"{checkpoints_namespace}/{model_name}"].upload(key)
+
+        # remove old models logged to experiment if they are not part of best k models at this point
+        if self.experiment.exists(checkpoints_namespace):
+            exp_structure = self.experiment.get_structure()
+            uploaded_model_names = self._get_full_model_names_from_exp_structure(
+                exp_structure,
+                checkpoints_namespace
+            )
+
+            for file_to_drop in list(uploaded_model_names - file_names):
+                del self.experiment[f"{checkpoints_namespace}/{file_to_drop}"]
+
+        # log best model path and best model score
+        if checkpoint_callback.best_model_path:
+            self.experiment[
+                self._construct_path_with_prefix("model/best_model_path")
+            ] = checkpoint_callback.best_model_path
+        if checkpoint_callback.best_model_score:
+            self.experiment[
+                self._construct_path_with_prefix("model/best_model_score")
+            ] = checkpoint_callback.best_model_score.cpu().detach().numpy()
+
+    @staticmethod
+    def _get_full_model_name(model_path: str, checkpoint_callback: "ReferenceType[ModelCheckpoint]") -> str:
+        """Returns model name which is string `modle_path` appended to `checkpoint_callback.dirpath`.
+        The problem here is that model name may contain slashes('/')."""
```

Comment:
Any idea how to resolve this ?

Reply:
It's resolved, the comment is explanation why such logic in `_get_full_model_name` was required at all, but it's misleading in fact.
I'll just remove this comment.